### PR TITLE
Initial documentation for GSoC 2024

### PR DESCRIPTION
**Pull request by AJun01**
This PR added a documentation page for GSoC 2024. Included anonymization techniques report for #77.

Updated OpenAPI(Swagger) Report for #81

### CODE REVIEW

**Review by iskitsas — COMMENTED** [review]
@AJun01 , You need to integrate this in docusaurus, did you try it?

File: doc/flexbench/docs/GSoC/2024.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+# 2024
+
+**Mentors**: Giannis Skitsas, Marios Karagiannopoulos
+
+**Assignee**: Yujun Liu
+
+## Abstract
+
+Flexbench was a forked project from the http-traffic-simulator npm package that has evolved significantly. The initial development during GSoC 2022 focused on transforming the package into a stand-alone desktop application and a web server with API exposure, supporting authentication and authorization through OpenIDC, and deploying it as a containerized application on Kubernetes. my work aims to further enhance Flexbench by implementing a anonymization feature to secure sensitive user data during testing, integrating OpenAPI to streamline and standardize API interactions, and addressing various UI/UX improvements for destop-app to ensure a more intuitive user experience.
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024.md:9]: Typo ". my work" should start with capital

**Comment by iskitsas** [comment]
It is better to create a new root item (folder) and place anything related with GSoC 2024 under that parent item

File: doc/flexbench/docs/GSoC/2024.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+<details>
+<summary>OpenAPI Integration Report</summary>
+
+[OpenAPI Integration Report](OpenAPI-Integration-Report.md)
+
+</details>
+
+## Implementation 
+
+### Destop-app
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024.md:36]: typo

File: doc/flexbench/docs/GSoC/2024.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+- Finding the bugs for aesthetic and UI/UX (S1)
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024.md:38]: What is (S1)?

Comment (AJun01) [reply doc/flexbench/docs/GSoC/2024.md:38]: S1 is something I picked up from the format of the last contributor in 2022.md, which matches the PR at the bottom, I will remove it if it is confusing.

**Comment by AJun01** [comment]
For the final documentation, I will create a new http://localhost:3000/flexbench/docs/GSoC/2024 and underneath this folder, I will create two section folders: Anonymization Feature&OpenAPI Feature, I will include everything related under them(README, instructions, installation, detailed information, etc. )

> It is better to create a new root item (folder) and place anything related with GSoC 2024 under that parent item
> 
> eg: http://localhost:3000/flexbench/docs/GSoC/2024/<your-docs>

**Comment by AJun01** [comment]
I have fixed the section for 2024, let me know if I can add anything to it to make anonymization and OpenAPI easier to understand

File: doc/flexbench/docs/GSoC/2024/2024.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- **Real-time Request and Response Anonymization**: Integrated anonymization directly into the request and response handling processes. This ensures that all data passing through the application is anonymized in real time, providing robust protection against accidental data leaks.
+- **Detailed Error Handling and Logging**: Included comprehensive error handling within the anonymization module to catch and log any issues that arise during the anonymization process. This ensures that the module can be debugged and maintained effectively.
+- **Documentation for Anonymization Module**: Provided thorough documentation on how to use and extend the anonymization module, including examples and best practices. This documentation serves as a guide for future developers to maintain and enhance the module.
+
+
+## Repository
+
+The original repository for this project can be found [here](https://github.com/flexivian/flexbench) 
+
+- All pull requests for GSoC 2021/2022/2024 can be found [here](https://github.com/flexivian/flexbench/pulls?q=).
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/2024.md:57]: You should only mention GSoC 2024

File: doc/flexbench/docs/GSoC/2024/2024.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+- For the implementation [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) was used and all Pull Requests target the develop branch.
+<!-- - All Pull Requests are squashed to one commit so that the history of the repository is cleaner. -->
+
+## Quick Start
+
+During the GSoC period work was done to easily set up a development environment and to quickly deploy the server app with docker-compose. A quickstart for both could be:
+
+- **Development**: Setup development environment as [described](../Installation/development.md).
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/2024.md:65]: The link is wrong. Redirect to `not found` page

File: doc/flexbench/docs/GSoC/2024/2024.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+- **Production**: Setup a quick production build with Docker as [described](../Installation/production.md).
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/2024.md:66]: The link is wrong. Redirect to `not found` page

File: doc/flexbench/docs/GSoC/2024/2024.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+In addition, the current site was built from scratch to support structured and extensive documentation. This site's source code is committed to the Git repository of this project.
+
+## Pull Requests
+
+### Aesthetic and UI/UX engancements [#75](https://github.com/flexivian/flexbench/pull/75)
+
+This PR fixed aesthetic and UI/UX for destop-app
+
+:::info
+- Finding the bugs for aesthetic and UI/UX (S1)/Fixing main process running bug
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/2024.md:77]: Remove (S1)

File: doc/flexbench/docs/GSoC/2024/Anonymization/README.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+let anonymizationEnabled = true;
+```
+
+## Usage
+
+### Enabling/Disabling Anonymization
+
+The anonymization process can be enabled or disabled using the following functions:
+
+```javascript
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/Anonymization/README.md:102]: The anonymization process is a feature flag of the core library. This code snippet here is something else? Are you describing the internal library? 

I would also expect here to see example of usage for the flexbench app, how to enable the anonymization feature. You should provide better docs here

Comment (iskitsas) [reply doc/flexbench/docs/GSoC/2024/Anonymization/README.md:102]: There is no section related on how to enable anonymization in flexbench

Comment (iskitsas) [reply doc/flexbench/docs/GSoC/2024/Anonymization/README.md:102]: How do I enable anonymization in server-app or desktop-app?

**Review by iskitsas — CHANGES_REQUESTED** [review]

File: doc/flexbench/docs/GSoC/2024/2024.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
+# 2024
+
+**Mentors**: Giannis Skitsas, Marios Karagiannopoulos
+
+**Assignee**: Yujun Liu
+
+## Abstract
+
+Flexbench, initially forked from the http-traffic-simulator npm package, has undergone significant enhancements. The initial development during GSoC 2022 transformed the package into a stand-alone desktop application and web server with API exposure, integrating OpenIDC for authentication and authorization and deploying it as a containerized application on Kubernetes. My work during GSoC 2024 further advances Flexbench by implementing a robust data anonymization feature, integrating OpenAPI support for streamlined API interactions, and addressing various UI/UX improvements for the desktop app to ensure a more intuitive user experience.
```
Comment (iskitsas) [inline doc/flexbench/docs/GSoC/2024/2024.md:9]: @AJun01  I do not remember integrating OpenIDC, did you find relevant implementation?

**Review by iskitsas — APPROVED** [review]

**Pull request merged**